src/server/run_sched.c

```diff
@@ -386,24 +388,29 @@ void
 scheduler_close(int sock)
 {
 	pbs_sched *psched;
-	int other_conn;
+	int other_conn = -1;
 
-	psched = find_sched_from_sock(sock);
-
-	if (psched == NULL)
-		return;
+	psched = find_sched_from_sock(sock, CONN_SCHED_PRIMARY);
+	if (psched == NULL) {
+		psched = find_sched_from_sock(sock, CONN_SCHED_SECONDARY);
```

Comment:
Consider adding a 3rd option of CONN_SCHED_ANY.  It'll save you from calling this twice.  Since the number of schedulers will be small, it isn't a big deal either way.